### PR TITLE
Implement full width sidebar

### LINKED ISSUE #1223: [Consumer UI Refresh] Replace Icon w/ Left arrow icon make "sidebar" take up the full width of the screen
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/master/CONTRIBUTING.md) for onboarding!
___
For this issue we must change the icon of the "hamburger" menu and do some light refactoring and styling.

- Remove all language in the code that refers to `hamburger` wherever `hamburger` is used we should replace it to `navigationMenu`
- Remove the hamburger menu icon itself from the [HeaderView](https://github.com/Expensify/ReactNativeChat/blob/9f48370459159c7599be6f23661de737319739a0/src/pages/home/HeaderView.js#L43-L52) and replace it with this chevron-left back style icon. Switch it to a `Pressable` instead of a `TouchableOpacity`.
- Make the "toggled open" state of the menu take up the entire width of the screen when viewing below the breakpoint.
- Remove the [`Pressable`](https://github.com/Expensify/Expensify.cash/blob/3f74c1f11e01a17039c539d4dcb0f81827d97e4e/src/pages/home/HomePage.js#L223-L230) that is listening for touches outside of the nav since the nav is now either pinned or full screen there is no need to listen for touches on an area that doesn’t exist. 
- Here is the asset we should use for the new icon: 
[arrow-left.svg.zip](https://github.com/Expensify/Expensify.cash/files/5799362/arrow-left.svg.zip)

**Note:** No proposal is required for this issue as we are asking contributors for exactly what we want.

# Screenshots

Screen Shot 2021-01-11 at 9 36 33 AM
Screen Shot 2021-01-11 at 9 37 15 AM

https://www.upwork.com/job/React-Native-Update-Refactoring-Task-Expensify-1223_~01f16b24a0fd3311c1/

### PR DESCRIPTION
**Pull request by marcaaron**
### Details
Migrating this over from 1244

### Fixed Issues
Fixes https://github.com/Expensify/Expensify.cash/issues/1223
Fixes https://github.com/Expensify/Expensify/issues/151430

### Tests
1. On mobile devices and small screens verify that the sidebar now takes up 100% of the width
2. At larger screen sizes verify the sidebar maintains the constant width of 300 pixels
3. Verify the hamburger icon has been replaced with a `<` arrow symbol.

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

### CODE REVIEW

**Comment by shawnborton** [comment]
From this mWeb screenshot here:
image


We actually want to get rid of the right border here as we only need the right border to create separation when the LHN is shown alongside a chat view on wider screens. I can't tell if iOS/Android has the right border too, but let's get rid of it there as well.

**Comment by marcaaron** [comment]
Nice catch @shawnborton. Cleaned up the styles and fixed the border issue.

File: src/pages/home/HeaderView.js
Comment (nickmurray47) [inline src/pages/home/HeaderView.js:44]: I don't think this is a blocker, but I noticed `Pressable` had a `hitSlop` property we could define that would make the back button easier to hit, especially on smaller devices. I noticed on the iOS emulator that the back button hit area didn't extend all the way to the left/topmost corner.

**Review by nickmurray47 — DISMISSED** [review]
Tests well on Web and iOS, and code LGTM

File: src/pages/home/HeaderView.js
Comment (marcaaron) [reply src/pages/home/HeaderView.js:44]: Cool do you know if this works on web? I started messing around with it. But it doesn't seem to do anything.

Comment (marcaaron) [reply src/pages/home/HeaderView.js:44]: Can definitely fix the issue you are describing. Just not sure if `hitSlop` is the way since it is not working on web.

**Comment by marcaaron** [comment]
Updated

**Comment by shawnborton** [comment]
Looks great, just a couple of testing notes that we might want to solve:
- On smaller viewports, there's still a shadow behind the LHN - I think we can kill that
- If the LHN is open and you resize your browser window, the LHN disappears and you go back to the chat window
- I wonder if 1000px for the breakpoint is a bit wide. Maybe we can back that down to something like 800? 


Things we might want to solve in a separate follow up PR:
- The animation of this feels strange to me. When viewing a chat and you want to navigate back to the LHN, I would think that the chat view would be the item animating and it would slide from left to right, and underneath it would appear the LHN. The way it works now is that when you tap the back button, the LHN appears over the content (and I understand that's because that's how the "old" LHN functioned, just noting it here for readers)
- So basically the behavior feels very unnatural to me. If you take WhatsApp or Slack for a spin, you'll see what I mean in comparison to how the navigation & animation is working here. I think it's important that the UX of this app feels natural, so I definitely think this is a problem worth solving.

**Comment by marcaaron** [comment]
>If the LHN is open and you resize your browser window, the LHN disappears and you go back to the chat window

I'm not really sure what you mean here. If you are viewing at a small window width and the browser is expanded the LHN disappears?

>I wonder if 1000px for the breakpoint is a bit wide. Maybe we can back that down to something like 800?

Yep we can try that. variables ftw!

>The animation of this feels strange to me. When viewing a chat and you want to navigate back to the LHN, I would think that the chat view would be the item animating and it would slide from left to right, and underneath it would appear the LHN. The way it works now is that when you tap the back button, the LHN appears over the content (and I understand that's because that's how the "old" LHN functioned, just noting it here for readers)

Interesting thought. We could do what you are asking. I think it would require making the report view slide in and out instead of the side bar. One idea (to sort of future proof this) is to explore using a navigation library. If we think that this is something we'll want to change again in the future then using something like [`react-navigation`](https://reactnavigation.org/) might give us more flexibility. Part of why we have what we have is because we don't have a "real navigation stack" and are just pushing different modals on top of each other. We can look into either just making the change you are asking for... or future proofing it so we can change our minds again in the future. But I agree it's for a separate PR.

**Comment by shawnborton** [comment]
> I'm not really sure what you mean here. If you are viewing at a small window width and the browser is expanded the LHN disappears?

You can easily reproduce this on your own:
- View the web app at a viewport that is small enough to make the LHN appear to be full screen
- When viewing the full screen LHN, adjust the browser width by just a little bit (not large enough to trigger desktop mode)
- Notice how the LHN disappears and slides out

**Comment by marcaaron** [comment]
Ok great thanks, I see it now. Trying something real quick.

**Comment by marcaaron** [comment]
Ok I think I got it sorted now.

**Comment by shawnborton** [comment]
Cool. I’ll likely be offline for the next 24 hours or so, so don’t let me
hold up the merge.

File: src/pages/home/HeaderView.js
Comment (nickmurray47) [reply src/pages/home/HeaderView.js:44]: Wow, works great now on the emulator. You're right in that adding to the padding is more robust and I think a better solution (previously thought that we could've done something like `hitSlop={{top: 10, left: 20, bottom: 10, right: 10}}` but it didn't change much)

**Review by nickmurray47 — DISMISSED** [review]

**Review by nickmurray47 — DISMISSED** [review]

**Comment by marcaaron** [comment]
Sorry @nickmurray47 this got conflicts again. Lmk if you are cool to merge this once it passes 🙃

**Review by nickmurray47 — APPROVED** [review]
Yeah, can do!

**Pull request merged**
